File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> numToIndex;
        vector<int> result;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums.at(i);

            if (numToIndex.find(complement) != numToIndex.end()) {
                result.push_back(numToIndex[complement]);
                result.push_back(i);
                return result;
            }
            else {
                numToIndex[nums.at(i)] = i;
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        
        result = []

        while left < right:
            numSum = numbers[left] + numbers[right]
            if numSum == target:
                result.append(left + 1)
                result.append(right + 1)
                return result
            elif numSum > target:
                right -= 1
            else:
                left += 1
        return result